src/net_processing.cpp

```diff
@@ -3913,13 +3913,11 @@ void PeerManager::EvictExtraOutboundPeers(int64_t time_in_seconds)
             AssertLockHeld(::cs_main);
 
             // Ignore non-outbound peers, or nodes marked for disconnect already
-            if (!pnode->IsOutboundOrBlockRelayConn() || pnode->fDisconnect) return;
+            if (!pnode->IsFullOutboundConn() || pnode->fDisconnect) return;
```

Comment:
To be clear, there was a bug here (no critical behavior, just meaningless code).
It was already useless to check for "BlockRelayConn", because they were filtered out later in "if (pnode->m_tx_relay == nullptr)". All good after this change.